frontend/src/metabase/parameters/utils/parameter-values.unit.spec.js

```diff
@@ -380,6 +346,52 @@ describe("parameters/utils/parameter-values", () => {
         "parameter2 default value",
       );
     });
+
+    describe("for number filter type", () => {
+      const numberParameter = {
+        id: 111,
+        slug: "numberParameter",
+        type: "number/=",
+      };
+
+      const runGetParameterValueFromQueryParams = value =>
+        getParameterValueFromQueryParams(
+          numberParameter,
+          {
+            [numberParameter.slug]: value,
+          },
+          metadata,
+        );
+
+      it("should parse the parameter value as a float when it is a number parameter without fields", () => {
+        expect(runGetParameterValueFromQueryParams("123.456")).toEqual([
+          123.456,
+        ]);
+      });
+
+      describe("when parsing parameter value that is a comma-separated list of numbers", () => {
+        it("should return list when every item is a number", () => {
+          expect(runGetParameterValueFromQueryParams("1, 2 ,   3")).toEqual([
+            "1,2,3",
+          ]);
+          expect(runGetParameterValueFromQueryParams("1,2.5,3.4")).toEqual([
+            "1,2.5,3.4",
+          ]);
+          expect(runGetParameterValueFromQueryParams("1,0,0000,2")).toEqual([
+            "1,0,0,2",
+          ]);
+        });
+
+        it("should return first float or NaN when list is not formatted properly", () => {
+          expect(runGetParameterValueFromQueryParams("1,,2,3,4")).toEqual([1]);
+          expect(runGetParameterValueFromQueryParams("1, ,2,3,4")).toEqual([1]);
+          expect(runGetParameterValueFromQueryParams("1,2,3,")).toEqual([1]);
+          expect(runGetParameterValueFromQueryParams(",,,")).toEqual([NaN]);
+          expect(runGetParameterValueFromQueryParams(" ")).toEqual([NaN]);
+          expect(runGetParameterValueFromQueryParams(",1,2,3")).toEqual([NaN]);
+        });
```

Comment:
Not sure we understood each other...
Most of the test cases I wrote in the comment were not expected. They were rather some edge cases that your parsing function does not cover.

I don't think we should ever return `NaN`, because it's then displayed in the filter widget. if there's no value (`" "`, `",,,"`), we should treat it as if no value has been passed from the query params.

If we have leading or trailing comma, there's no reason we shouldn't omit it (handle users' input mistake) and still parse the result.

Reply:
@nemanjaglumac Thank you, I've updated the implementation to handle the cases you mention 👍